middleware/index.js

```diff
@@ -0,0 +1,23 @@
+const db = require('../data/database-config')
+
+module.exports = {
+    logger: logger,
+    validateCarId: validateCarId,
+}
+
+
+function validateCarId(req, res, next) {
+    db('cars').where({ id: req.params.id })
+        .then(car => {
+            req.car = car[0];
+            next();
+        })
+        .catch(error => {
+            res.status(500).json({message: "There was an error getting the car." + " " + error})
+        })
+}
```

Comment:
Try using async await functions makes the code more readable and allows to limit nesting